pandas/tests/series/indexing/test_indexing.py

```diff
@@ -654,6 +654,106 @@ def test_timedelta_assignment():
     tm.assert_series_equal(s, expected)
 
 
+def test_timedelta_nat_assignment_series():
+    tdi = pd.timedelta_range("1s", periods=3, freq="s")
+    base = pd.Series(tdi)
+    expected = pd.Series([pd.NaT, tdi[1], tdi[2]], dtype="m8[ns]")
+
+    casting_nas = [pd.NaT, np.timedelta64("NaT", "ns")]
+    for nat in casting_nas:
+        ser = base.copy(deep=True)
+        ser[0] = nat
+        tm.assert_series_equal(ser, expected)
+
+        ser = base.copy(deep=True)
+        ser.loc[0] = nat
+        tm.assert_series_equal(ser, expected)
+
+        df = base.copy(deep=True).to_frame()
+        df.loc[0, 0] = nat
+        tm.assert_frame_equal(df, expected.to_frame())
+
+        df = base.copy(deep=True).to_frame()
+        df.iloc[0, 0] = nat
+        tm.assert_frame_equal(df, expected.to_frame())
+
+    # a specifically-datetime NaT should not be coerced to timedelta
+    expected = expected.astype(object)
```

Comment:
same

Reply:
Let's do #27323 first, as that is broken off from this and has a cleaner scope (and isn't red)